web/packages/design/src/DataTable/Pager/ClientSidePager/useClientSidePager.ts

```diff
@@ -0,0 +1,68 @@
+/*
+Copyright 2023 Gravitational, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { FetchStatus } from '../../types';
+
+export default function useClientSidePager({
+  data = [],
+  paginatedData = [],
+  currentPage,
+  pageSize,
```

Comment:
With `currentPage` and `pageSize` being optional starting on line 23 you could see an issue trying to access an `undefined` index. 

Can you add defaults for optional props.